### PR TITLE
Initial Zephyr version

### PR DESCRIPTION
**Pull request by magdalenaszumny**

### CODE REVIEW

File: zephyr/Kconfig [hunk 1]
```diff
@@ -0,0 +1,17 @@
+config LOBARO_COAP
```
Comment (qavaxb) [inline zephyr/Kconfig:1]:
```suggestion
menuconfig LOBARO_COAP
```

File: src/coap_options.c [hunk 2]
```diff
@@ -586,6 +586,8 @@ uint8_t CoAP_EncodeSzx(uint16_t blocksize) {
 			return 0b000;
 	}
 }
+
+#if defined(COAP_LL_INFO) || defined(COAP_LL_DEBUG)
```
Comment (qavaxb) [inline src/coap_options.c]:
```suggestion
#if defined(CONFIG_LOBARO_COAP_DEBUG_LEVEL) && (CONFIG_LOBARO_COAP_DEBUG_LEVEL > 1)
```

File: zephyr/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+    file(GLOB_RECURSE SOURCE_FILES ${CMAKE_CURRENT_LIST_DIR}/../src/*.h ${CMAKE_CURRENT_LIST_DIR}/../src/*.c)
+
+    zephyr_include_directories(${CMAKE_CURRENT_LIST_DIR}/../src)
+
+    zephyr_library_named(lobaro_coap)
+
+    zephyr_compile_definitions_ifdef(CONFIG_COAP_LL_DEBUG COAP_LL_DEBUG)
+    zephyr_compile_definitions_ifdef(CONFIG_COAP_LL_INFO COAP_LL_INFO)
+    zephyr_compile_definitions_ifdef(CONFIG_COAP_LL_ERROR COAP_LL_ERROR)
```
Comment (qavaxb) [inline zephyr/CMakeLists.txt]:
```suggestion

```

File: zephyr/Kconfig [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    depends on LOBARO_COAP
+    default y
+
+config COAP_LL_INFO
+    bool "INFO logs"
+    depends on LOBARO_COAP
+
+config COAP_LL_DEBUG
+    bool "DEBUG logs"
+    depends on LOBARO_COAP
```
Comment (qavaxb) [inline zephyr/Kconfig]:
```suggestion

if LOBARO_COAP

config LOBARO_COAP_DEBUG_LEVEL
	int "Lobaro Coap debug level"
	default 1
	range 0 3
	help
	  Default lobaro_coap debug logging level for Zephyr integration code
	  0 No debug
	  1 Error
	  2 Information
	  3 Verbose
  
   endif #LOBARO_COAP
```

File: src/coap_options.c (hunk 2 above)
Comment (tgujda) [reply src/coap_options.c]: done

File: zephyr/CMakeLists.txt (hunk 3 above)
Comment (tgujda) [reply zephyr/CMakeLists.txt]: done

File: zephyr/Kconfig (hunk 4 above)
Comment (tgujda) [reply zephyr/Kconfig]: done

**Comment by magdalenaszumny** [comment]
Looks great! I cannot approve, because my commit is included... @qavaxb - could you please approve?

**Review by qavaxb — APPROVED** [review]
Ok

**Pull request merged**